lib/DOMFetcher.js

```diff
@@ -0,0 +1,29 @@
+/* globals Services */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/Services.jsm");
+
+const FRAMESCRIPT_PATH = "resource://loci/data/DocumentChangeMessenger.js";
```

Comment:
do not hardcode this path.

Use this snippet:
```javascript
const simplePrefs = require("sdk/simple-prefs");
const options = require("@loader/options");
const baseURI = simplePrefs["sdk.baseURI"] || options.prefixURI;
function getResourceURL(path) {
  return `${baseURI}${path}`;
}

const FRAMESCRIPT_PATH = getResourceURL("/data/DocumentChangeMessenger.js");
```

You could move this utility function in something like `lib/utils.js`